distribution/tools/upgrade-cli/src/main/java/org/opensearch/upgrade/InstallPluginsTask.java

```diff
@@ -28,27 +56,44 @@ public void accept(final Tuple<TaskInput, Terminal> input) {
             return;
         }
         terminal.println("Installing core plugins ...");
-        final ProcessBuilder processBuilder = new ProcessBuilder();
+        List<String> manualPlugins = new ArrayList<>();
+
         for (String plugin : taskInput.getPlugins()) {
-            // TODO - validate if plugin is an official plugin for OpenSearch.
-            final String command = taskInput.getOpenSearchBin().resolve("opensearch-plugin") + " install " + plugin;
-            if (OS.WINDOWS == OS.current()) {
-                processBuilder.command("cmd.exe", "/c", command);
+            if (OFFICIAL_PLUGINS.contains(plugin)) {
+                executeInstallPluginCommand(plugin, taskInput, terminal);
             } else {
-                processBuilder.command("sh", "-c", command);
-            }
-            try {
-                final Process process = processBuilder.inheritIO().start();
-                if (process.waitFor() != 0) {
-                    terminal.errorPrint(Terminal.Verbosity.NORMAL, String.format(Locale.getDefault(), ERROR_MSG, plugin));
-                }
-            } catch (IOException | InterruptedException e) {
-                terminal.errorPrint(Terminal.Verbosity.NORMAL, String.format(Locale.getDefault(), ERROR_MSG, plugin) + e.getMessage());
+                manualPlugins.add(plugin);
             }
         }
+        if (!manualPlugins.isEmpty()) {
+            terminal.println("Please install the following custom plugins manually: " + manualPlugins);
+        }
         terminal.println("Success!" + System.lineSeparator());
     }
 
+    public void executeInstallPluginCommand(String plugin, TaskInput taskInput, Terminal terminal) {
+        ProcessBuilder processBuilder = getCommandsBasedOnOS(plugin, taskInput);
+        try {
+            final Process process = processBuilder.inheritIO().start();
+            if (process.waitFor() != 0) {
+                terminal.errorPrint(Terminal.Verbosity.NORMAL, String.format(Locale.getDefault(), ERROR_MSG, plugin));
+            }
+        } catch (IOException | InterruptedException e) {
+            terminal.errorPrint(Terminal.Verbosity.NORMAL, String.format(Locale.getDefault(), ERROR_MSG, plugin) + e.getMessage());
+        }
+    }
+
+    public ProcessBuilder getCommandsBasedOnOS(String plugin, TaskInput taskInput) {
```

Comment:
This returns a ProcessBuilder - the method name suggests it should return a command string.

Reply:
Renamed the method.